jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/gzip/GzipResponseAndCallback.java

```diff
@@ -167,27 +164,27 @@ protected void commit(boolean last, Callback callback, ByteBuffer content)
         if (LOG.isDebugEnabled())
             LOG.debug("commit(last={}, callback={}, content={})", last, callback, BufferUtil.toDetailString(content));
 
-        // Are we excluding because of status?
         Request request = getRequest();
         Response response = this;
+        HttpFields.Mutable fields = response.getHeaders();
```

Comment:
If we wanted to be a tiny bit more efficient here, we could iterate over all the fields once and extract `ETAG`, `CONTENT_TYPE`, `CONTENT_ENCODING`, and `CONTENT_LENGTH` in a single loop.

But perhaps the number of response fields is not large enough (and iterators cheap enough) that this is not worth it.

Reply:
I wrote a JMH benchmark, and iteration is so darn cheap that the extra cost of using the iterator takes a lot to amortize.

It takes a `ImmutableHttpFields` instance containing 20 fields to make looking up the last 4 fields of the internal array cheaper with the iterator.

Looking up less than 4 fields, or having to iterate less than 20 entries per lookup makes the solution using `get` always faster.